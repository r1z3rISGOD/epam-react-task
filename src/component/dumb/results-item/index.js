import { ResultsItem } from './results-item'
export { ResultsItem }
