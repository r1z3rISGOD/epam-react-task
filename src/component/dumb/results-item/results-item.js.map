{"version":3,"file":"results-item.js","sourceRoot":"","sources":["results-item.tsx"],"names":[],"mappings":";;;AAAA,6BAA8B;AAC9B,+BAA4B;AAYrB,IAAM,WAAW,GAAgC,UAAC,EAQxD;QAPC,KAAK,WAAA,EACL,MAAM,YAAA,EACN,IAAI,UAAA,EACJ,OAAO,aAAA,EACP,EAAE,QAAA,EACF,MAAM,YAAA,EACN,YAAY,kBAAA;IAEZ,OAAO,CACD,6BAAK,SAAS,EAAC,MAAM,EAAC,OAAO,EAAE,cAAM,OAAA,YAAY,CAAC,EAAE,EAAE,MAAM,CAAC,EAAxB,CAAwB;QACzD,6BAAK,SAAS,EAAC,WAAW,EAAC,GAAG,EAAE,OAAO,EAAE,GAAG,EAAE,KAAK,GAAG;QACtD,6BAAK,SAAS,EAAC,WAAW;YACtB,6BAAK,SAAS,EAAC,iBAAiB;gBAC5B,8BAAM,SAAS,EAAC,kBAAkB,IAAE,KAAK,CAAC,WAAW,EAAE,CAAQ;gBAC/D,8BAAM,SAAS,EAAC,mBAAmB,IAAE,MAAM,CAAC,CAAC,CAAC,CAAQ,CACpD;YACN,8BAAM,SAAS,EAAC,iBAAiB,IAAE,IAAI,CAAC,MAAM,CAAC,CAAC,EAAE,CAAC,CAAC,CAAQ,CAC1D,CACJ,CACX,CAAA;AACH,CAAC,CAAA;AArBY,QAAA,WAAW,eAqBvB"}