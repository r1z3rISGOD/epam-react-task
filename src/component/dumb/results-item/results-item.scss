@import "src/_variables/variables";

.item {
  width: 30%;
  display: flex;
  flex-direction: column;
  margin-bottom: 50px;
}

.item__img {
  width: 100%;
  height: 90%;
}

.item-data__text {
  display: flex;
  flex-direction: column;
}

.item-data {
  display: flex;
  justify-content: space-between;
  font-family: $font;
}

.item-data__year {

}

.item-data__title {

}

.item-data__genres {

}
