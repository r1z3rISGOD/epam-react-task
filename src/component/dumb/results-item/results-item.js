import React from 'react'
import PropTypes from 'prop-types'
import './results-item.scss'
import { putOpenedFilmToStore, putOpenedFilmsGenreToStore, putFetchedArrayToStore } from '../../../store/actions'
import { useDispatch } from 'react-redux'
import { useHistory } from 'react-router-dom'
import { fetchData, fetchFilm } from '../../services/fetchService'

export const ResultsItem = (props) => {
  const history = useHistory()
  const dispatch = useDispatch()
  const { title, genres, year, picture, id, active } = props
  return (
        <div className="item" onClick={async () => {
          const fetchedFilm = await fetchFilm(id)
          dispatch(putOpenedFilmToStore(fetchedFilm))
          dispatch(putOpenedFilmsGenreToStore(fetchedFilm.genres[0]))
          history.push(`/films/${fetchedFilm.id}`)
          const newFetchedArray = await fetchData(fetchedFilm.genres[0], 'genres', active)
          dispatch(putFetchedArrayToStore(newFetchedArray.data))
        }}>
            <img className="item__img" src={picture} alt={title}/>
            <div className="item-data">
                <div className='item-data__text'>
                    <span className="item-data__title">{title.toUpperCase()}</span>
                    <span className="item-data__genres">{genres[0]}</span>
                </div>
                <span className="item-data__year">{year.substr(0, 4)}</span>
            </div>
        </div>
  )
}

ResultsItem.propTypes = {
  title: PropTypes.string,
  genres: PropTypes.array,
  year: PropTypes.string,
  id: PropTypes.number,
  picture: PropTypes.string
}
