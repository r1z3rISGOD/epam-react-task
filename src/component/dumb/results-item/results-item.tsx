import * as React from 'react'
import './results-item.scss'

interface ResultsItemProps {
    title: string,
    genres: any[],
    year: string,
    picture: string,
    id: number,
    active : string,
    fetchingFilm: any
}

export const ResultsItem : React.FC<ResultsItemProps> = ({
  title,
  genres,
  year,
  picture,
  id,
  active,
  fetchingFilm
}) => {
  return (
        <div className="item" onClick={() => fetchingFilm(id, active)}>
            <img className="item__img" src={picture} alt={title}/>
            <div className="item-data">
                <div className='item-data__text'>
                    <span className="item-data__title">{title.toUpperCase()}</span>
                    <span className="item-data__genres">{genres[0]}</span>
                </div>
                <span className="item-data__year">{year.substr(0, 4)}</span>
            </div>
        </div>
  )
}
