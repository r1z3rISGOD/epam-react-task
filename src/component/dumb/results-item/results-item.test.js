import { ResultsItem } from './results-item.tsx'
import React from 'react'

const film = {
  id: 412302,
  title: 'Gemini',
  tagline: '',
  vote_average: 10,
  vote_count: 3,
  release_date: '2018-03-30',
  poster_path: 'https://image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg',
  overview: 'A heinous crime tests the complex relationship between a tenacious personal assistant and her Hollywood starlet boss. As the assistant travels across Los Angeles to unravel the mystery, she must stay one step ahead of a determined policeman and confront her own understanding of friendship, truth and celebrity.',
  budget: 0,
  revenue: 0,
  genres: [
    'Mystery',
    'Thriller'
  ],
  runtime: 92
}

const mockFetchingFilm = jest.fn((id, active) => { console.log(id, active) })

it('results item', () => {
  const comp = shallow(<ResultsItem active='genres' fetchingFilm={mockFetchingFilm} genres={film.genres} id={film.id} picture={film.poster_path} title={film.title} year={film.release_date}/>)
  expect(comp).toMatchSnapshot()
  const btn = comp.find('.item')
  btn.simulate('click')
  expect(mockFetchingFilm.mock.calls.length).toBe(1)
})
