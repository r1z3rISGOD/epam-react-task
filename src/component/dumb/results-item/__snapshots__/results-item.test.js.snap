// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`results item 1`] = `
<div
  className="item"
  onClick={[Function]}
>
  <img
    alt="Gemini"
    className="item__img"
    src="https://image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg"
  />
  <div
    className="item-data"
  >
    <div
      className="item-data__text"
    >
      <span
        className="item-data__title"
      >
        GEMINI
      </span>
      <span
        className="item-data__genres"
      >
        Mystery
      </span>
    </div>
    <span
      className="item-data__year"
    >
      2018
    </span>
  </div>
</div>
`;
